environment/config.go

```diff
@@ -160,16 +160,19 @@ func (c Configuration) Configure(cfgParams ConfigurationParameters) (ConfigureRe
 
 		nsInitStatic := namespace.NewStaticInitializer(nsList)
 
-		shardSet, hostShardSets, err := newStaticShardSet(c.Static.TopologyConfig.Shards, c.Static.ListenAddress, cfgParams.HostID)
+		shardSet, hostShardSets, err := newStaticShardSet(c.Static.TopologyConfig.Shards, c.Static.TopologyConfig.Hosts)
 		if err != nil {
 			err = fmt.Errorf("unable to create shard set for static config: %v", err)
 			return emptyConfig, err
 		}
 		staticOptions := topology.NewStaticOptions().
-			SetReplicas(1).
 			SetHostShardSets(hostShardSets).
 			SetShardSet(shardSet)
 
+		if c.Static.TopologyConfig.Replicas != 0 {
```

Comment:
As discussed, let's default to replicas 0 and make sure replicas always matches `len(hosts)` or else users are going to have a very strange experience, because we only support static topology for clusters where each host holds the full set of data with this current config setup.